api/dataset.go

```diff
@@ -14,53 +14,57 @@ import (
 )
 
 func (api *DatasetAPI) getDatasets(w http.ResponseWriter, r *http.Request) {
-	if err := api.auditor.Record(r.Context(), getDatasetsAction, actionAttempted, nil); err != nil {
-		handleAuditingFailure(w, err, nil)
+	ctx := r.Context()
+	if err := api.auditor.Record(ctx, getDatasetsAction, actionAttempted, nil); err != nil {
+		auditActionFailure(ctx, getDatasetsAction, actionAttempted, err, nil)
+		handleErrorType(datasetDocType, errs.ErrAuditActionAttemptedFailure, w)
 		return
 	}
 
-	results, err := api.dataStore.Backend.GetDatasets()
-	if err != nil {
-		log.Error(err, nil)
-		if auditErr := api.auditor.Record(r.Context(), getDatasetsAction, actionUnsuccessful, nil); auditErr != nil {
-			handleAuditingFailure(w, auditErr, nil)
-			return
-		}
-		handleErrorType(datasetDocType, err, w)
-		return
-	}
-
-	authorised, logData := api.authenticate(r, log.Data{})
-
-	var b []byte
-	if authorised {
-
-		// User has valid authentication to get raw dataset document
-		datasets := &models.DatasetUpdateResults{}
-		datasets.Items = results
-		b, err = json.Marshal(datasets)
+	b, err := func() ([]byte, error) {
+		results, err := api.dataStore.Backend.GetDatasets()
 		if err != nil {
-			log.ErrorC("failed to marshal dataset resource into bytes", err, nil)
-			handleErrorType(datasetDocType, err, w)
-			return
+			log.Error(err, nil)
```

Comment:
Would it be worth changing this back to a context error and use the request id in `ctx` for contextual part of log?

`log.ErrorC( requestID.Get(ctx), err, nil )`